tests/e2e_tests/conftest.py

```diff
@@ -65,50 +70,109 @@ def authenticator_fund_config(request: FixtureRequest) -> TestFundConfig:
+
+        else:  # This flow is used in CI/CD where credentials are available implicitly
+            ssm_client = boto3.client("ssm")
+            rsa256_private_key_base64 = ssm_client.get_parameter(Name=param_name, WithDecryption=True)["Parameter"][
+                "Value"
+            ]
+        rsa256_private_key = base64.b64decode(rsa256_private_key_base64)
+        del rsa256_private_key_base64
+
+    return rsa256_private_key
+
+
+@pytest.fixture
+def context(request: FixtureRequest, browser: Browser):
+    e2e_env = request.config.getoption("e2e_env")
+    return browser.new_context(
+        **(
+            dict(
+                http_credentials={
+                    "username": Config.E2E_DEVTEST_BASIC_AUTH_USERNAME,
+                    "password": Config.E2E_DEVTEST_BASIC_AUTH_PASSWORD,
+                }
+            )
+            if e2e_env in {"dev", "test"}
+            else {}
+        )
+    )
+
+
 @pytest.fixture()
 def user_auth(
     request: FixtureRequest,
     domains: FundingServiceDomains,
     authenticator_fund_config: TestFundConfig,
-    page: Page,
+    context: BrowserContext,
 ) -> Account:
+    e2e_env = request.config.getoption("e2e_env")
+    e2e_aws_vault_profile = request.config.getoption("e2e_aws_vault_profile")
+
     email_address = generate_email_address(
         test_name=request.node.originalname,
         email_domain="communities.gov.uk",
     )
 
-    user_roles = request.node.get_closest_marker("user_roles")
-
-    account = create_account_with_roles(
-        email_address=email_address,
-        roles=user_roles.args[0] if user_roles else [],
+    roles_marker = request.node.get_closest_marker("user_roles")
+    user_roles = roles_marker.args[0] if roles_marker else []
+
+    now = int(datetime.datetime.timestamp(datetime.datetime.now()))
+    jwt_data = {
+        "accountId": str(uuid.uuid4()),
+        "azureAdSubjectId": str(uuid.uuid4()),
+        "email": email_address,
+        "fullName": f"E2E Test User - {request.node.originalname}",
+        "roles": user_roles,
+        "iat": now,
+        "exp": now + (15 * 60),  # 15 minutes from now
+    }
+
+    def create_token(payload):
+        algorithm = "RS256"  # Must match the algorithm used by fsd-authenticator
+        return jwt.encode(payload, _read_jwt_signing_key(e2e_env, e2e_aws_vault_profile), algorithm=algorithm)
+
+    context.add_cookies(
+        [
+            {
+                "name": Config.FSD_USER_TOKEN_COOKIE_NAME,
+                "value": create_token(jwt_data),
+                "domain": domains.cookie,
+                "path": "/",
+                "httpOnly": True,
+                "secure": True,
+                "sameSize": "Strict",
+            }
+        ]
     )
 
-    response = requests.get(f"{domains.authenticator}/magic-links")
-    magic_links_before = set(response.json())
-
-    new_magic_link_page = NewMagicLinkPage(page, domain=domains.authenticator, fund_config=authenticator_fund_config)
```

Comment:
You can probably remove the authentication.py MagicLink classes.

Reply:
good shout